code/modules/modular_computers/computers/item/modular_computer.dm

```diff
@@ -0,0 +1,813 @@
+	var/obj/item/weapon/computer_hardware/nano_printer/nano_printer					// Nano Printer component of this computer, for your everyday paperwork needs.
+	var/obj/item/weapon/computer_hardware/hard_drive/portable/portable_drive		// Portable data storage
+
+	var/list/idle_threads = list()							// Idle programs on background. They still receive process calls but can't be interacted with.
+	var/obj/physical = null
+
+
+
+// Eject ID card from computer, if it has ID slot with card inside.
+/obj/item/modular_computer/verb/eject_id()
+	set name = "Eject ID"
+	set category = "Object"
+	set src in view(1)
+
+	if(usr.incapacitated() || !istype(usr, /mob/living))
+		usr << "<span class='warning'>You can't do that.</span>"
+		return
+
+	if(!Adjacent(usr))
+		usr << "<span class='warning'>You can't reach it.</span>"
+		return
+
+	proc_eject_id(usr)
+
+// Eject ID card from computer, if it has ID slot with card inside.
+/obj/item/modular_computer/verb/eject_usb()
+	set name = "Eject Portable Device"
+	set category = "Object"
+	set src in view(1)
+
+	if(usr.incapacitated() || !istype(usr, /mob/living))
+		usr << "<span class='warning'>You can't do that.</span>"
+		return
+
+	if(!Adjacent(usr))
+		usr << "<span class='warning'>You can't reach it.</span>"
+		return
+
+	proc_eject_usb(usr)
+
+/obj/item/modular_computer/proc/proc_eject_id(mob/user)
+	if(!user)
+		user = usr
+
+	if(!card_slot)
+		user << "\The [src] does not have an ID card slot"
+		return
+
+	if(!card_slot.stored_card)
+		user << "There is no card in \the [src]"
+		return
+
+	if(active_program)
+		active_program.event_idremoved(0)
+
+	for(var/I in idle_threads)
+		var/datum/computer_file/program/P = I
+		P.event_idremoved(1)
+
+	card_slot.stored_card.forceMove(get_turf(src))
+	card_slot.stored_card = null
+//	update_uis()
+	user << "You remove the card from \the [src]"
+
+/obj/item/modular_computer/proc/proc_eject_usb(mob/user)
+	if(!user)
+		user = usr
+
+	if(!portable_drive)
+		user << "There is no portable device connected to \the [src]."
+		return
+
+	uninstall_component(user, portable_drive)
+//	update_uis()
+
+/obj/item/modular_computer/attack_ghost(mob/dead/observer/user)
+	if(enabled)
+		ui_interact(user)
+	else if(check_rights_for(user.client, R_ADMIN))
+		var/response = alert(user, "This computer is turned off. Would you like to turn it on?", "Admin Override", "Yes", "No")
+		if(response == "Yes")
+			turn_on(user)
+
+/obj/item/modular_computer/emag_act(mob/user)
+	if(computer_emagged)
+		user << "\The [src] was already emagged."
+		return 0
+	else
+		computer_emagged = 1
+		user << "You emag \the [src]. It's screen briefly shows a \"OVERRIDE ACCEPTED: New software downloads available.\" message."
+		return 1
+
+/obj/item/modular_computer/examine(mob/user)
+	..()
+	if(damage > broken_damage)
+		user << "<span class='danger'>It is heavily damaged!</span>"
+	else if(damage)
+		user << "It is damaged."
+
+/obj/item/modular_computer/New()
+	machines += src
```

Comment:
Something probably assumes machinery is just machines. This is not the best solution.